includes/Core/Util/User_Input_Settings.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\User_Input_Settings
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Google\Site_Kit\Core\Storage\Transients;
+use Google\Site_Kit\Core\Storage\User_Transients;
+
+/**
+ * Class managing requests to user input settings endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class User_Input_Settings {
+
+	/**
+	 * Authentication object.
+	 *
+	 * @since n.e.x.t
+	 * @var Authentication
+	 */
+	private $authentication;
+
+	/**
+	 * Transients object.
+	 *
+	 * @since n.e.x.t
+	 * @var Transients
+	 */
+	private $transients;
+
+	/**
+	 * User_Transients object.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Transients
+	 */
+	private $user_transients;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context         $context         Plugin context.
+	 * @param Authentication  $authentication  Optional. Authentication instance. Default a new instance.
+	 * @param Transients      $transients      Optional. Transient API instance. Default a new instance.
+	 * @param User_Transients $user_transients Optional. User Transients API instance. Default a new instance.
+	 */
+	public function __construct(
+		Context $context,
+		Authentication $authentication = null,
+		Transients $transients = null,
+		User_Transients $user_transients = null
+	) {
+		$this->transients      = $transients ?: new Transients( $context );
+		$this->user_transients = $user_transients ?: new User_Transients( $context );
+		$this->authentication  = $authentication ?: new Authentication( $context, null, null, $this->transients );
+	}
+
+	/**
+	 * Determines whether the site is connected to proxy or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean TRUE if connected, otherwise FALSE.
+	 */
+	private function is_connected_to_proxy() {
+		if ( ! $this->authentication->is_authenticated() ) {
+			return false;
+		}
+
+		$credentials = $this->authentication->credentials();
+		if ( $credentials ) {
+			if ( ! $credentials->using_proxy() || ! $credentials->has() ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sends POST request to the proxy's settings endpoint to sync user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array User input settings.
```

Comment:
The return value can also be a `WP_Error`.

Reply:
Updated.